test/extended/openstack/loadbalancers.go

```diff
@@ -419,3 +437,59 @@ func createTestDeployment(depName string, labels map[string]string, replicas int
 	}}
 	return testDeployment
 }
+
+// Active wait (up to 3 minutes) until a certain number of members match a regex on status attribute.
+func waitUntilNmembersReady(client *gophercloud.ServiceClient, pool *pools.Pool, n int32, status string) error {
+
+	var resultingMembers []string
+	var err error
+
+	e2e.Logf("Waiting for %d members online in the pool %s", n, pool.Name)
+	err = wait.Poll(15*time.Second, 3*time.Minute, func() (bool, error) {
+		resultingMembers, err = getMembersOnStatus(client, pool, status)
+		if err != nil {
+			return false, err
+		}
+		if len(resultingMembers) == int(n) {
+			e2e.Logf("Found expected number of %s members, checking again 5 seconds later to confirm that it is stable", status)
+			time.Sleep(5 * time.Second)
+			resultingMembers, err = getMembersOnStatus(client, pool, status)
+			if err != nil {
+				return false, err
+			} else if len(resultingMembers) == int(n) {
+				e2e.Logf("Wait completed, %s members found: %q", status, resultingMembers)
+				return true, nil
+			}
+		}
+		e2e.Logf("Waiting for next iteration, %s members found: %q", status, resultingMembers)
+		return false, nil
+	})
+
+	if err != nil && strings.Contains(err.Error(), "timed out waiting for the condition") {
+		allMembers, err := pools.ListMembers(client, pool.ID, pools.ListMembersOpts{}).AllPages()
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("unexpected number of %s after active wait. All members status: %q", status, allMembers)
+	}
+	return err
+}
+
+// Return the name of the members matching a regex in the OperatingStatus attribute
+func getMembersOnStatus(client *gophercloud.ServiceClient, pool *pools.Pool, status string) ([]string, error) {
+
+	var members []string
+	for _, aux := range pool.Members {
+		member, err := pools.GetMember(client, pool.ID, aux.ID).Extract()
+		if err != nil {
+			return nil, err
```

Comment:
Don't you need to return an empty list instead?
```suggestion
			return [], err
```

Reply:
According to this https://blog.boot.dev/golang/empty-vs-nil-slices-golang/, `nil` is better